doc/Learn/Tutorials/Markup/HowTo-CustomMarkupProject-VisualStates.md

```diff
@@ -0,0 +1,848 @@
+
+		    <VisualStateManager.VisualStateGroups>
+			    <VisualStateGroup x:Name="ButtonStates">
+				    <VisualState x:Name="CustomPointerEntered">
+					    <VisualState.Setters>
+						    <Setter Target="visualStateButtonChips.Width" Value="500" />
+					    </VisualState.Setters>
+				    </VisualState>
+				    <VisualState x:Name="CustomPointerExited">
+					    <VisualState.Setters>
+						    <Setter Target="visualStateButtonChips.Width" Value="200" />
+					    </VisualState.Setters>
+				    </VisualState>
+			    </VisualStateGroup>
+		    </VisualStateManager.VisualStateGroups>
+		
+		    <utu:ChipGroup x:Name="chipGroup" Grid.Row="0">
+			    <utu:Chip Margin="5" Content="Chip 1" Background="LightBlue" />
+			    <utu:Chip Margin="5" Content="Chip 2">
+				    <utu:Chip.Style>
+					    <Style TargetType="utu:Chip">
+						    <Setter Property="Foreground" Value="Red" />
+					    </Style>
+				    </utu:Chip.Style>
+			    </utu:Chip>
+			    <utu:Chip Margin="5" Content="Chip 3" x:Name="chipElement" Checked="chip_Checked" Unchecked="chip_Unchecked"/>
+
+		    </utu:ChipGroup>
+
+		    <Button Grid.Row="1" Content="Visual State on UserControl" x:Name="visualStateButtonChips"/>
+
+	    </Grid>
+    </UserControl>
+    ```
+
+    - MainPage.xaml.cs
+
+    ```csharp
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices.WindowsRuntime;
+    using Windows.Foundation;
+    using Windows.Foundation.Collections;
+    using Windows.UI.Xaml;
+
+
+    // The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+    namespace MySampleToolkitProjectXAML
+    {
+	    public sealed partial class SampleUserControl : UserControl
+	    {
+		    public SampleUserControl()
+		    {
+			    this.InitializeComponent();
+			    Loaded += SampleUserControl_Loaded;
+		    }
+
+		    private void SampleUserControl_Loaded(object sender, RoutedEventArgs e)
+		    {
+			    visualStateButtonChips.PointerEntered += (sender, e) =>
+			    {
+				    VisualStateManager.GoToState(this, "CustomPointerEntered", true);
+			    };
+
+			    visualStateButtonChips.PointerExited += (sender, e) =>
+			    {
+				    VisualStateManager.GoToState(this, "CustomPointerExited", true);
+			    };
+		    }
+
+		    public void chip_Unchecked(object sender, RoutedEventArgs e)
+		    {
+			    chipElement.FontSize = 14;
+		    }
+		    public void chip_Checked(object sender, RoutedEventArgs e)
+		    {
+			    chipElement.FontSize = 18;
+		    }
+	    }
+    }
+
+    ```
+
+## Try it yourself
+
+Now try to change your MainPage to have different layout and test other attributes and elements.
+
+## Next Steps
+
+- [Custom your own C# Markup - Learn how to Change the Theme](xref:Learn.Tutorials.HowToCustomMarkupProjectTheme)
+- [Custom your own C# Markup - Learn how to use MVUX](xref:Learn.Tutorials.HowToCustomMarkupProjectMVUX)
```

Comment:
```suggestion
    ```csharp
    
	visualStateButtonChips.PointerEntered += (sender, e) =>
	{
		VisualStateManager.GoToState(this, "PointerEntered", true);
	};

	visualStateButtonChips.PointerExited += (sender, e) =>
	{
		VisualStateManager.GoToState(this, "PointerExited", true);
	};


    ```

    # [**XAML**](#tab/cli)
    
    #### XAML

    
    ##### Add VisualStateManager to the Page.

    In the Shared Project open the file *MainPage.xaml* and change the content to have the VisualStateManager.

    After the start of the first StackPanel add the XAML below

    ```xml
		<VisualStateManager.VisualStateGroups>
			<VisualStateGroup x:Name="ButtonStates">
				<VisualState x:Name="CustomPointerEntered">
					<VisualState.Setters>
						<Setter Target="btn.Width" Value="500" />
						<Setter Target="navigationButton.Background" Value="Blue" />
					</VisualState.Setters>
				</VisualState>
				<VisualState x:Name="CustomPointerExited">
					<VisualState.Setters>
						<Setter Target="btn.Width" Value="200" />
						<Setter Target="navigationButton.Background" Value="White" />
					</VisualState.Setters>
				</VisualState>
			</VisualStateGroup>
		</VisualStateManager.VisualStateGroups>
    ```

    We will change the navigationButton (the same that has been used for controlling the Navigation) and will the Background Color now.
    And change the value of the Width of the Button that will be used on the EventHandler.
    You can add the Button before the `<utu:ChipGroup x:Name="chipGroup">`.

    ```csharp
	<Button  x:Name="btn" Content="VisualStateManager Test" Width="300"/>
    ```

    To handle the Event Handlers we need to create the Event Handlers so that the events are fired and the actions happen.
    For that open the MainPage.xaml.cs and add the Event Handlers.

    ```csharp
	public MainPage()
	{
		this.InitializeComponent();
		Loaded += MainPage_Loaded;
	}

	private void MainPage_Loaded(object sender, RoutedEventArgs e)
	{
		btn.PointerEntered += (sender, e) =>
		{
			VisualStateManager.GoToState(this, "CustomPointerEntered", true);
		};

		btn.PointerExited += (sender, e) =>
		{
			VisualStateManager.GoToState(this, "CustomPointerExited", true);
		};
	}

    ```

    Now Run the Project and test it.

    ##### Add VisualStateManager to the UserControl.

    In the Shared Project open the file *SampleUserControl.xaml* and change the content to have the VisualStateManager.
    For the User Control, we will use the Grid Element to attach the VisualStateManager.

    ```csharp
    new Grid()
	.RowDefinitions<Grid>("Auto, *")
	.Children(
    ...
    )
    .VisualStateManager(builder =>
				builder.Group("ButtonStates",
					groupBuilder =>
						groupBuilder
						.State("PointerEntered",
						stateBuilder => stateBuilder
							.Setters(visualStateButtonChips, e => e.Width(500))
						).State("PointerExited",
							stateBuilder => stateBuilder
								.Setters(visualStateButtonChips, e => e.Width(200))
						)
					)
			)
    ```

    We will change the value of the Width of the Button that will be used on the EventHandler.
    You can add the Button in the Second Row of the new Grid.

    ```xml
    <Button Grid.Row="1" Content="Visual State on UserControl" x:Name="visualStateButtonChips"/>
    ```

    To handle the Event Handlers we need to create the Event Handlers so that the events are fired and the actions happen.

    ```csharp
    public SampleUserControl()
	{
		this.InitializeComponent();
		Loaded += SampleUserControl_Loaded;
	}

	private void SampleUserControl_Loaded(object sender, RoutedEventArgs e)
	{
		visualStateButtonChips.PointerEntered += (sender, e) =>
		{
			VisualStateManager.GoToState(this, "CustomPointerEntered", true);
		};

		visualStateButtonChips.PointerExited += (sender, e) =>
		{
			VisualStateManager.GoToState(this, "CustomPointerExited", true);
		};
	}
    ```


    # [**Full Code**](#tab/code)

    #### Full C# Markup code
    
    - Example of the complete code on the MainPage.cs, so you can follow along in your own project.

    ```csharp
   using Microsoft.UI;
    using Uno.Toolkit.UI;

    namespace MySampleToolkitProject;

    public sealed partial class MainPage : Page
    {
	    public MainPage()
	    {
		    this
		        .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
			    .Content(
				    new StackPanel()
					    .Children(
						    new NavigationBar().Content("Title Main Page")
							    .VerticalAlignment(VerticalAlignment.Top)
							    .HorizontalAlignment(HorizontalAlignment.Left),
						    new StackPanel()
							    .Margin(0,50,0,0)
							    .VerticalAlignment(VerticalAlignment.Center)
							    .HorizontalAlignment(HorizontalAlignment.Center)
							    .Children(
								    new Button()
									    .Content("Go to Second Page")
									    .Assign(out var navigationButton),
								    new Button()
									    .Width(200)
									    .Height(40)
									    .Content("VisualStateManager Test")
									    .Assign(out var btn),
								    new SampleUserControl()
							    )
					    )
					    .VisualStateManager(builder =>
							    builder.Group("ButtonStates",
								    groupBuilder =>
									    groupBuilder
									    .State("PointerEntered",
										    stateBuilder => stateBuilder
											    .Setters(btn, e => e.Width(500))
											    .Setters(navigationButton, e => e.Background(new SolidColorBrush(Colors.Blue)))
									    ).State("PointerExited",
										    stateBuilder => stateBuilder
											    .Setters(btn, e => e.Width(200))
											    .Setters(navigationButton, e => e.Background(new SolidColorBrush(Colors.White)))
									    )
								    )
						    )
			    );

		    navigationButton.Click += (s, e) =>
		    {
			    Frame.Navigate(typeof(SecondPage));
		    };

		    btn.PointerEntered += (sender, e) =>
		    {
			    VisualStateManager.GoToState(this, "PointerEntered", true);
		    };

		    btn.PointerExited += (sender, e) =>
		    {
			    VisualStateManager.GoToState(this, "PointerExited", true);
		    };

	    }
    }

    ```
    #### Full XAML code
    
    - MainPage.xaml

    ```xml
    <Page x:Class="MySampleToolkitProjectXAML.MainPage"
	    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
	    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	    xmlns:local="using:MySampleToolkitProjectXAML"
	    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
	    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
	    mc:Ignorable="d"
	    xmlns:utu="using:Uno.Toolkit.UI"
	    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
	
	    <StackPanel>

		    <VisualStateManager.VisualStateGroups>
			    <VisualStateGroup x:Name="ButtonStates">
				    <VisualState x:Name="CustomPointerEntered">
					    <VisualState.Setters>
						    <Setter Target="btn.Width" Value="500" />
						    <Setter Target="navigationButton.Background" Value="Blue" />
					    </VisualState.Setters>
				    </VisualState>
				    <VisualState x:Name="CustomPointerExited">
					    <VisualState.Setters>
						    <Setter Target="btn.Width" Value="200" />
						    <Setter Target="navigationButton.Background" Value="White" />
					    </VisualState.Setters>
				    </VisualState>
			    </VisualStateGroup>
		    </VisualStateManager.VisualStateGroups>

		    <utu:NavigationBar Content="Title Main Page"
					    HorizontalAlignment="Left"
					    VerticalAlignment="Top"/>
		    <StackPanel HorizontalAlignment="Center"
			    VerticalAlignment="Center">

			    <Button Content="Go to Second Page" x:Name="navigationButton" Click="navigationButton_Click"/>
			
			    <Button  x:Name="btn" Content="VisualStateManager Test" Width="300"/>

			    <utu:ChipGroup x:Name="chipGroup">
				    <utu:Chip Margin="5" Content="Chip 1" Background="LightBlue" />
				    <utu:Chip Margin="5" Content="Chip 2">
					    <utu:Chip.Style>
						    <Style TargetType="utu:Chip">
							    <Setter Property="Foreground" Value="Red" />
						    </Style>
					    </utu:Chip.Style>
				    </utu:Chip>
				    <utu:Chip Margin="5" Content="Chip 3" x:Name="chipElement" Checked="chip_Checked" Unchecked="chip_Unchecked"/>

			    </utu:ChipGroup>
		    </StackPanel>
	    </StackPanel>
    </Page>

    ```

    - MainPage.xaml.cs

    ```csharp
    using Windows.Devices.Input;

    namespace MySampleToolkitProjectXAML;

    public sealed partial class MainPage : Page
    {
	    public MainPage()
	    {
		    this.InitializeComponent();
		    Loaded += MainPage_Loaded;
	    }

	    private void MainPage_Loaded(object sender, RoutedEventArgs e)
	    {
		    btn.PointerEntered += (sender, e) =>
		    {
			    VisualStateManager.GoToState(this, "CustomPointerEntered", true);
		    };

		    btn.PointerExited += (sender, e) =>
		    {
			    VisualStateManager.GoToState(this, "CustomPointerExited", true);
		    };
	    }

	    public void navigationButton_Click(object sender, RoutedEventArgs e)
	    {
		    Frame.Navigate(typeof(SecondPage));
	    }
	    public void chip_Unchecked(object sender, RoutedEventArgs e)
	    {
		    chipElement.FontSize = 14;
	    }
	    public void chip_Checked(object sender, RoutedEventArgs e)
	    {
		    chipElement.FontSize = 18;
	    }
    }

    ```

    - SampleUserControl.xaml

    ```xml
    <UserControl
        x:Class="MySampleToolkitProjectXAML.SampleUserControl"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:local="using:MySampleToolkitProjectXAML"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d"
	    xmlns:utu="using:Uno.Toolkit.UI"
        d:DesignHeight="300"
        d:DesignWidth="400">

        <Grid>
		    <Grid.RowDefinitions>
			    <RowDefinition Height="Auto" />
			    <RowDefinition Height="*" />
		    </Grid.RowDefinitions>

		    <VisualStateManager.VisualStateGroups>
			    <VisualStateGroup x:Name="ButtonStates">
				    <VisualState x:Name="CustomPointerEntered">
					    <VisualState.Setters>
						    <Setter Target="visualStateButtonChips.Width" Value="500" />
					    </VisualState.Setters>
				    </VisualState>
				    <VisualState x:Name="CustomPointerExited">
					    <VisualState.Setters>
						    <Setter Target="visualStateButtonChips.Width" Value="200" />
					    </VisualState.Setters>
				    </VisualState>
			    </VisualStateGroup>
		    </VisualStateManager.VisualStateGroups>
		
		    <utu:ChipGroup x:Name="chipGroup" Grid.Row="0">
			    <utu:Chip Margin="5" Content="Chip 1" Background="LightBlue" />
			    <utu:Chip Margin="5" Content="Chip 2">
				    <utu:Chip.Style>
					    <Style TargetType="utu:Chip">
						    <Setter Property="Foreground" Value="Red" />
					    </Style>
				    </utu:Chip.Style>
			    </utu:Chip>
			    <utu:Chip Margin="5" Content="Chip 3" x:Name="chipElement" Checked="chip_Checked" Unchecked="chip_Unchecked"/>

		    </utu:ChipGroup>

		    <Button Grid.Row="1" Content="Visual State on UserControl" x:Name="visualStateButtonChips"/>

	    </Grid>
    </UserControl>
    ```

    - MainPage.xaml.cs

    ```csharp
    using System;
    using System.Collections.Generic;
    using System.IO;
    using System.Linq;
    using System.Runtime.InteropServices.WindowsRuntime;
    using Windows.Foundation;
    using Windows.Foundation.Collections;
    using Windows.UI.Xaml;


    // The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236

    namespace MySampleToolkitProjectXAML
    {
	    public sealed partial class SampleUserControl : UserControl
	    {
		    public SampleUserControl()
		    {
			    this.InitializeComponent();
			    Loaded += SampleUserControl_Loaded;
		    }

		    private void SampleUserControl_Loaded(object sender, RoutedEventArgs e)
		    {
			    visualStateButtonChips.PointerEntered += (sender, e) =>
			    {
				    VisualStateManager.GoToState(this, "CustomPointerEntered", true);
			    };

			    visualStateButtonChips.PointerExited += (sender, e) =>
			    {
				    VisualStateManager.GoToState(this, "CustomPointerExited", true);
			    };
		    }

		    public void chip_Unchecked(object sender, RoutedEventArgs e)
		    {
			    chipElement.FontSize = 14;
		    }
		    public void chip_Checked(object sender, RoutedEventArgs e)
		    {
			    chipElement.FontSize = 18;
		    }
	    }
    }

    ```

## Try it yourself

Now try to change your MainPage to have a different layout and test other attributes and elements.

## Next Steps

- [Custom your own C# Markup - Learn how to Change the Theme](xref:Learn.Tutorials.HowToCustomMarkupProjectTheme)
- [Custom your own C# Markup - Learn how to use MVUX](xref:Learn.Tutorials.HowToCustomMarkupProjectMVUX)
```